### PR TITLE
Make click dragging an item onto a stored HUD equip it

### PR DESCRIPTION
**Pull request by FlameArrow57**
[GAME OBJECTS][QoL]

## About the PR 
This PR makes it so that click dragging an item stored in a storage HUD onto the HUD location where it's stored will equip it

## Why's this needed? 
When you are trying really fast to equip an item, it's common that instead of just clicking the item, you will click the item, hold left mouse button, and release when it's on the HUD where it's stored which won't equip it, making you think you never clicked it or some bug happened. This makes it so that doing so will actually equip it instead of doing nothing.

## Changelog 
```changelog
(u)FlameArrow57
(+)Improved item equip consistency when equipping an item from your inventory with fast mouse flicks.
```

### CODE REVIEW

File: code/datums/hud/storage.dm [hunk 1]
```diff
@@ -105,6 +60,70 @@
...
+			return
+		if (!(O in src.master.get_contents()))
+			return
+		var/clicked_position = src.get_clicked_position(user, params)
+		if (!clicked_position)
+			return
+		if (src.obj_locs[clicked_position] == O)
+			user.click(O, params2list(params))
+
+	proc/get_clicked_position(mob/user, list/params)
```
Comment (TobleroneSwordfish) [inline code/datums/hud/storage.dm:74]: doc comment pls

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
may as well make it work for unequipping too for consistency

**Comment by FlameArrow57** [comment]
Can you clarify? Unequipping is done with hotkey, or just click drag to a single inventory spot?

**Comment by TobleroneSwordfish** [comment]
Click dragging to hand didn't seem to work for unequipping? I know it's specifically coded for backpacks but it makes sense to be general if we're doing this.

**Comment by FlameArrow57** [comment]
I see what you're saying, but point of the PR was to just fix the one issue that's in the description, that's a different thing

**Pull request merged**